### PR TITLE
Mat 5591 export bundle from repo 2

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE PR

Jira Ticket: [MAT-5591](https://jira.cms.gov/browse/MAT-5591)
(Optional) Related Tickets:

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    } catch (NullPointerException nullPointerException) {
+      throw new InternalServerException("Resource name cannot be null");
+    }
+  }
+
+  /**
+   * @param bundleResource Bundle resource
+   * @return r4 resource
+   */
+  public static Resource getResource(Bundle bundleResource, String resourceType) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java:31]: Having both a system `Resource` and a FHIR `Resource` type in the same Resource Utility confused me for a time.

File: src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.hl7.fhir.r4.model.Resource;
+
+import cms.gov.madie.measure.exceptions.InternalServerException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+
+public class ResourceUtils {
+  public static String getData(String resource) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java:14]: Having a system resource and a fhir resource type in the same utility confused me for a bit.

Maybe the `getData` method can live in the `ResourceUtility` interface as a default method?

File: src/main/java/cms/gov/madie/measure/services/BundleService.java [hunk 3]
```diff
@@ -58,19 +67,55 @@ public ResponseEntity<byte[]> exportBundleMeasure(Measure measure, String access
...
-        throw new InvalidResourceBundleStateException(
-            "Measure", measure.getId(), "since there are no associated developers in metadata.");
-      } else if (measure.getMeasureMetaData().getSteward() == null) {
-        throw new InvalidResourceBundleStateException(
-            "Measure", measure.getId(), "since there is no associated steward in metadata.");
-      } else if (StringUtils.isBlank(measure.getMeasureMetaData().getDescription())) {
+    isMetadataValid(measure);
+    areGroupsValid(measure);
+    try {
+      retrieveElmJson(measure, accessToken);
```
Comment (jkotanchik-SB) [inline src/main/java/cms/gov/madie/measure/services/BundleService.java]: Noted `retrieveElmJson` is called under the `draft` branch below. 
Does it need to be called for versioned measures?

File: src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtilityImpl.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            .filter(
+                entry ->
+                    StringUtils.equalsIgnoreCase(
+                        "Measure", entry.getResource().getResourceType().toString()))
+            .findFirst();
+    org.hl7.fhir.r4.model.DomainResource measure =
+        (org.hl7.fhir.r4.model.DomainResource)
+            measureEntry
+                .map(org.hl7.fhir.r4.model.Bundle.BundleEntryComponent::getResource)
+                .orElse(null);
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtilityImpl.java]:
```suggestion
    org.hl7.fhir.r4.model.DomainResource measure =
            (DomainResource) ResourceUtils.getResource(bundle, "Measure");
```
Nearly identical.

File: src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java (hunk 2 above)
Comment (gregory-akins) [reply src/main/java/gov/cms/madie/measure/utilities/qicore411/ResourceUtils.java:14]: Maybe

Prolly

File: src/main/java/cms/gov/madie/measure/services/BundleService.java (hunk 3 above)
Comment (gregory-akins) [reply src/main/java/cms/gov/madie/measure/services/BundleService.java]: Guess not.. it should be in the bundle already.

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**